devenv/src/command.rs

```diff
@@ -140,7 +140,9 @@ impl App {
             };
 
             if self.cli.impure || self.config.impure {
-                flags.push("--impure");
+                if !command.starts_with("nix-store") {
```

Comment:
maybe we should instead allow it only when `nix develop`?

Reply:
Maybe? Do any of the `build` or `eval` commands need impure?

Reply:
yeah, maybe we should do it when `command.ends_with("nix")`

Reply:
or `command == "nix"` at that point?